### ISSUE TITLE
Can't locate element using accessibility_id on iOS

### ISSUE THREAD

**Issue by rrudiuk-devx** [appium/appium#14992]
## The problem

I'm trying to use accessibility ids to locate elements on iOS.

I'm using these two requests (with Python):

self.driver.find_elements_by_accessibility_id('Welcome')
self.driver.find_elements_by_xpath('//XCUIElementTypeButton[@name="Welcome"]')

For xpath I get the element located everytime and for accessibility id I always get an error.

## Environment

* Appium version 1.20.0-beta.0
* Desktop OS/version used to run Appium: MacOS Big Sur 11.1
* Node.js version (unless using Appium.app|exe): v15.3.0
* Npm or Yarn package manager: npm
* Mobile platform/version under test: 14.3 and 14.4
* Real device or emulator/simulator: real device
* Appium CLI or Appium.app|exe: Appium.app (installed as a desktop app)

## Details

I'm new to iOS automation (previously dealt with Android) and just searching for a suitable location strategy. So I wonder if I'm doing something wrong or that's a bug. Couldn't find any similar reports over the web.

## Link to Appium logs

[WD Proxy] Matched '/elements' to command name 'findElements'
[WD Proxy] Proxying [POST /elements] to [POST http://127.0.0.1:8100/session/6A639E72-F4A6-46E1-B9A2-A6C8240ADA1A/elements] with body: {"using":"accessibility id","value":"Welcome"}
[WD Proxy] Got response with status 200: {"value":[],"sessionId":"6A639E72-F4A6-46E1-B9A2-A6C8240ADA1A"}
[W3C (8ec766f6)] Responding to client with driver.findElements() result: []

**Comment by mykola-mokhnach** [comment]
This could be a race condition issue. Usually xpath lookup needs more time for execution than other lookup types. Make sure the element with the given name is really present on the page first.

**Comment by rrudiuk-devx** [comment]
Thanks! So I did the following:

I have a separate file where I initiate methods. And there I created two separate methods:

class WelcomePage(BasePage):
    def should_be_correct_welcome_title_x(self):
        assert self.driver.find_elements_by_xpath('//XCUIElementTypeButton[@name="Welcome"]'), "Title not found with xpath"

    def should_be_correct_welcome_title_a(self):
        assert self.driver.find_elements_by_accessibility_id('Welcome'), "Title not found with accessibility id"

Then I call them both in the same test where I call the method that uses xpath before the one with accessibility id and add an additional delay between.

    def test_should_be_welcome_screen(self, driver):
        welcome_page = WelcomePage(driver)
        welcome_page.should_be_correct_welcome_title_x()
        time.sleep(2)
        welcome_page.should_be_correct_welcome_title_a()

And I still get this: AssertionError: Title not found with accessibility id

**Comment by rrudiuk-devx** [comment]
@mykola-mokhnach thanks, using your suggestion I was able to identify what the problem was.

In short, my accessibility id is actually not "Welcome", but something like "Welcome to our great app" (I can't provide the exact title as the app is already in production. Anyways, sorry I didn't mention that. I had no idea that this may cause any issues). And in the app it actually looks like this:

Welcome to our
great app

And it appears that I have to use accessibility id "Welcome to our\ngreat app".

Not really comfortable as this will require more testing to ensure that I've entered the correct id, but it works much faster. So I guess I'll stick with it.

Thanks for the help!
